src/rock_physics_open/equinor_utilities/machine_learning_utilities/exponential_model.py

```diff
@@ -1,119 +1,133 @@
+from __future__ import annotations
+
 import pickle
-from typing import Union
+from typing import Any, Dict, Union
 
 import numpy as np
 
+from .base_pressure_model import BasePressureModel
 
-def _verify_input(inp_arr):
-    if isinstance(inp_arr, np.ndarray) and not (
-        inp_arr.ndim == 2 and inp_arr.shape[1] == 3
-    ):
-        raise ValueError(
-            "Input to predict method should be an nx3 numpy array with columns velocity, in situ "
-            "pressure and depleted pressure"
-        )
 
+class ExponentialPressureModel(BasePressureModel):
+    """
+    Exponential pressure sensitivity model for velocity prediction.
+
+    Uses exponential decay function: v = v0 * (1 - a*exp(-p/b)) / (1 - a*exp(-p0/b))
+    where v0 is reference velocity, p is pressure, a and b are model parameters.
+
+    Input format (n,3): [velocity, p_eff_in_situ, p_eff_depleted]
+    """
 
-class CarbonateExponentialPressure:
     def __init__(
         self,
-        a_factor: float = None,
-        b_factor: float = None,
-        model_max_pressure: float = None,
+        a_factor: float,
+        b_factor: float,
+        model_max_pressure: float | None = None,
         description: str = "",
     ):
+        """
+        Initialize exponential pressure model.
+
+        Parameters
+        ----------
+        a_factor : float
+            Exponential amplitude parameter [unitless].
+        b_factor : float
+            Exponential decay parameter [Pa].
+        model_max_pressure : float | None
+            Maximum pressure for predict_max method [Pa].
+        description : str
+            Model description.
+        """
+        super().__init__(model_max_pressure, description)
         self._a_factor = a_factor
         self._b_factor = b_factor
-        self._model_max_pressure = model_max_pressure
-        self._description = description
-
-    def todict(self):
-        return {
-            "a_factor": self._a_factor,
-            "b_factor": self._b_factor,
-            "model_max_pressure": self._model_max_pressure,
-            "description": self._description,
-        }
 
     @property
     def a_factor(self) -> float:
+        """Exponential amplitude factor."""
         return self._a_factor
 
     @property
     def b_factor(self) -> float:
+        """Exponential decay factor."""
         return self._b_factor
 
-    @property
-    def max_pressure(self) -> float:
-        return self._model_max_pressure
+    def validate_input(self, inp_arr: np.ndarray) -> np.ndarray:
```

Comment:
Tip: When overriding the methods of a base class, it can be a good idea to use an override decorator from the `typing` library.


```suggestion
    @override
    def validate_input(self, inp_arr: np.ndarray) -> np.ndarray:
```

This way we make it clear that this method is supposed to be an override of the abstract method from the baseclass. For example, if one uses the wrong name in the method of the derived class, one could get unintentional behavoir.

This tip applies for all methods that override abstract methods of the baseclass